connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/UplynkSsaiDescription.kt

```diff
@@ -0,0 +1,89 @@
+package com.theoplayer.android.connector.uplynk
+
+import com.theoplayer.android.api.source.ssai.CustomSsaiDescription
+import kotlinx.serialization.Serializable
+
+/**
+ * The configuration for server-side ad insertion using the [UplynkConnector].
+ */
+@Serializable
+class UplynkSsaiDescription(
+    val prefix: String?,
+    val assetIds: List<String>,
+    val externalId: List<String>,
+    val userId: String?,
+    val preplayParameters: LinkedHashMap<String, String>
```

Comment:
Weaken this to a generic `Map<String, String>`.

Reply:
I made this LinkedHashMap to maintain an order of preplay parameters. This is needed to support signed urls. https://docs.edgecast.com/video/#Setup/Playback-URLs.htm#SigningPURLs

In the previous implementation we had indeed `preplayParameters` and `orderedParameters` separately. I think it was because we introduced `preplayParameters` and only then thought about signed urls. Our orderedParameters were like this:

```
        @NonNull
        public Builder orderedParameters(@NonNull LinkedHashMap<String, String> orderedParameters) {
            this.preplayParameters = orderedParameters;
            return this;
        }
```

So I decided to put LinkedHashMap in `preplayParameters` to avoid duplication. Do you think we need both `preplayParameters` and `orderedParameters` in this connector?

Reply:
Hmm, fair enough. I suppose requiring a `LinkedHashMap` is the easiest way to enforce that the map preserves the order of its entries.